### PR TITLE
Xex loader: Fixes for imports, XEX1 support

### PR DESCRIPTION
**Pull request by smx-smx**

### CODE REVIEW

File: src/ImageLoaders/XexLoader/XexLoader.cs [hunk 1]
```diff
@@ -223,43 +228,20 @@ private void LoadHeaders()
                         XexImportLibraryBlockHeader blockHeader = new StructureReader<XexImportLibraryBlockHeader>(rdr).Read();
 
                         long string_table = rdr.Offset;
-
-                        rdr.Offset += blockHeader.string_table_size;
+                        for(int _i=0; _i<blockHeader.count; _i++) {
```
Comment (uxmal) [inline src/ImageLoaders/XexLoader/XexLoader.cs]: Consider using `j` rather than `_i`.

File: src/ImageLoaders/XexLoader/XexLoader.cs [hunk 2]
```diff
@@ -90,10 +92,13 @@ private void LoadHeaders()
 
             XexHeader header = xexData.header;
 
-            if (header.xex2 != XEX2_MAGIC)
-            {
-                throw new BadImageFormatException("Invalid XEX Magic");
-            }
+			switch (header.magic) {
```
Comment (uxmal) [inline src/ImageLoaders/XexLoader/XexLoader.cs]: Please conform to the formatting used in the project: `{` on new line.

File: src/ImageLoaders/XexLoader/XexLoader.cs [hunk 3]
```diff
@@ -603,7 +595,7 @@ private void PopulateImports()
                 byte type = (byte)((value & 0xFF000000) >> 24);
                 byte libIndex = (byte)((value & 0x00FF0000) >> 16);
 
-                if(type == 0) {
+                if(type == 0) { //variable
```
Comment (uxmal) [inline src/ImageLoaders/XexLoader/XexLoader.cs]: Consider defining a few constants for this:
```
private const int DataImport = 0;
private const int FunctionImport = 1;
```

**Comment by smx-smx** [comment]
Done

**Comment by uxmal** [comment]
Looks good now.

File: src/ImageLoaders/XexLoader/Structures.cs [hunk 4]
```diff
@@ -28,307 +28,340 @@
 
 namespace Reko.ImageLoaders.Xex
 {
-    public class Structures
-    {
-        public const UInt32 XEX2_MAGIC = 0x58455832; //XEX2
+	public class Structures
+	{
+		public const UInt32 XEX1_MAGIC = 0x58455831; //XEX1
```
Comment (ptomin) [inline src/ImageLoaders/XexLoader/Structures.cs]: These are tabs. Should not be spaces?

Comment (smx-smx) [reply src/ImageLoaders/XexLoader/Structures.cs]: i'm struggling with VS2019 not honoring .editorconfig for existing files
i had to use VSCode with an extension to reformat the code...
i don't think it was doing this with VS2017

**Pull request merged**